tests/test-ray-tracing-hitobject-intrinsics.cpp

```diff
@@ -433,3 +472,106 @@ GPU_TEST_CASE("ray-tracing-hitobject-make-hit", ALL | DontCreateDevice)
     ComPtr<ISlangBlob> resultBlob = test.getTestResult();
     checkQueryAndInvokeResult(resultBlob);
 }
+
+GPU_TEST_CASE("ray-tracing-hitobject-make-miss", ALL)
+{
+    if (!device->hasFeature(Feature::RayTracing))
+        SKIP("ray tracing not supported");
+    if (!device->hasFeature(Feature::ShaderExecutionReordering))
+        SKIP("shader execution reordering not supported");
+
+    RayTracingSingleTriangleTest test;
+    test.init(device);
+    test.createResultBuffer(sizeof(TestResult));
+    test.run("test-ray-tracing-hitobject-intrinsics", "rayGenShaderMakeMiss", {"closestHitNOP"}, {"missInvoke"});
+
+    ComPtr<ISlangBlob> resultBlob = test.getTestResult();
+    checkQueryAndInvokeResult(resultBlob);
+}
+
+GPU_TEST_CASE("ray-tracing-hitobject-make-motion-miss", ALL)
+{
+    if (!device->hasFeature(Feature::RayTracing))
+        SKIP("ray tracing not supported");
+    if (!device->hasFeature(Feature::ShaderExecutionReordering))
+        SKIP("shader execution reordering not supported");
+    if (!device->hasFeature(Feature::RayTracingMotionBlur))
+        SKIP("ray tracing motion blur not supported");
+
+    RayTracingSingleTriangleMotionTest test;
+    test.init(device);
+    test.createResultBuffer(sizeof(TestResult));
+    test.run(
+        "test-ray-tracing-hitobject-intrinsics",
+        "rayGenShaderMakeMotionMiss",
+        {"closestHitNOP"},
+        {"missInvoke"},
+        RayTracingPipelineFlags::EnableMotion
+    );
+
+    ComPtr<ISlangBlob> resultBlob = test.getTestResult();
+    checkQueryAndInvokeResult(resultBlob);
+}
+
+GPU_TEST_CASE("ray-tracing-hitobject-make-motion-hit", ALL | DontCreateDevice)
+{
+    // Limit the shader model to SM 6.6 for this test, since the NVAPI headers don't support MakeHit
+    // for newer shader models.
+    DeviceExtraOptions extraOptions;
+    extraOptions.d3d12HighestShaderModel = 0x66; // SM 6.6
+    device = createTestingDevice(ctx, ctx->deviceType, false, &extraOptions);
+    REQUIRE(device);
+
+    if (!device->hasFeature(Feature::RayTracing))
+        SKIP("ray tracing not supported");
+    if (!device->hasFeature(Feature::ShaderExecutionReordering))
+        SKIP("shader execution reordering not supported");
+    if (!device->hasFeature(Feature::RayTracingMotionBlur))
+        SKIP("ray tracing motion blur not supported");
+
+    // Disabled under pre OptiX 9.0 due to https://github.com/shader-slang/slang/issues/8723
+    if (device->getDeviceType() == DeviceType::CUDA && device->getInfo().optixVersion < 90000)
```

Comment:
Is https://github.com/shader-slang/slang/issues/8723 fixable? I.e. can we support SER with OptiX 8.1 at all? If not, we should just require OptiX 9.0 for SER and remove this check.